tests/acceptance/01_vars/02_functions/validjson_trailing_bogus_data.cf

```diff
@@ -0,0 +1,41 @@
+##############################################################################
+#
+# Test validjson() evaluates to !any:: when there is trailing bogus data after
+# termination for JSON object.
+#
+##############################################################################
+
+body common control
+{
+  bundlesequence => { "test", "check" };
+  version => "1.0";
+}
+
+##############################################################################
+
+bundle agent test
+{
+  meta:
+      "description" -> { "CFE-4080" }
+        string => "Test validjson() with trailing bogus data";
+}
+
+##############################################################################
+
+bundle agent check
+{
+  classes:
+      "ok"
+        and => {
+          not(validjson('"some": [ "json" ] }')),
+          not(validjson('{ "some": [ "json" ] } [')),
```

Comment:
I would add a couple more cases:

1. The trailing part is just a letter, like `a`
2. The trailing part itself is valid JSON, for example `validjson('{"blah": "foo"} {}')`